cpp/velox/memory/VeloxMemoryManager.h

```diff
@@ -0,0 +1,153 @@
+    }
+    return freeBytes;
+  }
+
+  bool growMemory(
+      velox::memory::MemoryPool* pool,
+      const std::vector<std::shared_ptr<velox::memory::MemoryPool>>& candidatePools,
+      uint64_t targetBytes) override {
+    GLUTEN_CHECK(candidatePools.size() == 1, "ListenableArbitrator should only be used within a single root pool");
+    auto candidate = candidatePools.back();
+    GLUTEN_CHECK(pool->root() == candidate.get(), "Illegal state in ListenableArbitrator");
+    growPool(pool, targetBytes);
+    return true;
+  }
+
+  Stats stats() const override {
+    Stats stats; // no-op
+    return stats;
+  }
+
+  std::string toString() const override {
+    return fmt::format(
+        "ARBITRATOR[{}] CAPACITY {} {}", kindString(kind_), velox::succinctBytes(capacity_), stats().toString());
+  }
+
+ private:
+  void growPool(memory::MemoryPool* pool, uint64_t bytes) {
+    listener_->allocationChanged(bytes);
+    pool->grow(bytes);
+  }
+
+  void abort(velox::memory::MemoryPool* pool) {
+    try {
+      pool->abort();
+    } catch (const std::exception& e) {
+      LOG(WARNING) << "Failed to abort memory pool " << pool->toString();
+    }
+    // NOTE: no matter memory pool abort throws or not, it should have been marked
+    // as aborted to prevent any new memory arbitration triggered from the aborted
+    // memory pool.
+    VELOX_CHECK(pool->aborted());
+  }
+
+  gluten::AllocationListener* listener_;
+};
+
+class VeloxMemoryManager final : public MemoryManager {
+ public:
+  explicit VeloxMemoryManager(
+      std::string name,
+      std::shared_ptr<MemoryAllocator> allocator,
+      std::shared_ptr<AllocationListener> listener)
+      : MemoryManager(), name_(name) {
+    glutenAlloc_ = std::make_shared<ListenableMemoryAllocator>(allocator.get(), listener);
+
+    velox::memory::MemoryArbitrator::Config arbitratorConfig{
+        velox::memory::MemoryArbitrator::Kind::kNoOp, // do not use shared arbitrator as it will mess up the thread
+                                                      // contexts (one Spark task reclaims memory from another)
+        velox::memory::kMaxMemory, // the 2nd capacity
+        0,
+        32 << 20,
+        true};
+    velox::memory::IMemoryManager::Options mmOptions{
+        velox::memory::MemoryAllocator::kMaxAlignment,
+        velox::memory::kMaxMemory, // the 1st capacity, Velox requires for a couple of different capacity numbers
+        true,
+        false,
+        velox::memory::MemoryAllocator::getInstance(),
+        [=]() { return std::make_unique<ListenableArbitrator>(arbitratorConfig, listener.get()); },
+    };
+    veloxMemoryManager_ = std::make_unique<velox::memory::MemoryManager>(mmOptions);
+    veloxPool_ = veloxMemoryManager_->addRootPool(
+        name_ + "_root",
+        velox::memory::kMaxMemory, // the 3rd capacity
+        facebook::velox::memory::MemoryReclaimer::create());
+    veloxLeafPool_ = veloxPool_->addLeafChild(name_ + "_default_leaf");
+  }
```

Comment:
Any particular reason moving this code and ListenableArbitrator out from `VeloxMemoryPool.cc`?

Reply:
Just make related codes more closer, and these are declared in cc file before, thus not export symbol to other.

Reply:
and VeloxMemoryPool.h should be removed, lots of codes use `defaultLeafVeloxMemoryPool`, but this not tracked by Spark.

Reply:
Or I'll suggest (based on the code before this patch)

1. rename `VeloxMemoryPool.h/cc` to `VeloxMemoryManager.h/cc`
2. in `VeloxMemoryManager.h`, remove the old functions, and add declaration of class `class VeloxMemoryManager final : public MemoryManager` to it
3. in `VeloxMemoryManager.cc`, add the class definition of `class VeloxMemoryManager final : public MemoryManager`